packages/zapp/console/src/components/Executions/ExecutionDetails/ExecutionNodeViews.tsx

```diff
@@ -146,19 +144,20 @@ export const ExecutionNodeViews: React.FC<ExecutionNodeViewsProps> = ({ executio
     );
   };
 
-  const renderTab = (tabType) => (
-    <WaitForQuery
-      errorComponent={DataError}
-      query={childGroupsQuery}
-      loadingComponent={LoadingComponent}
-    >
-      {() => <ExecutionTab tabType={tabType} abortMetadata={abortMetadata ?? undefined} />}
-    </WaitForQuery>
-  );
-
-  if (loading) {
-    return <LoadingComponent />;
-  }
+  const renderTab = (tabType) => {
+    if (loading) {
+      return <LoadingComponent />;
+    }
+    return (
+      <WaitForQuery
+        errorComponent={DataError}
+        query={childGroupsQuery}
+        loadingComponent={LoadingComponent}
```

Comment:
if the WaitForQuery component already takes in a loading component, why do we need the extra check on line 148?